packages/vscode-extension/src/codeLensProvider.ts

```diff
@@ -98,16 +116,62 @@ export class ManifestTemplateCodeLensProvider implements vscode.CodeLensProvider
   private manifestConfigDataRegex = /{{config.manifest[\.a-zA-Z]+}}/g;
   private manifestStateDataRegex = /{{state\.[a-zA-Z-_]+\.\w+}}/g;
 
+  private projectConfigs: ProjectConfigV3 | undefined = undefined;
+  private mutex = new Mutex();
+
   public provideCodeLenses(
     document: vscode.TextDocument
   ): vscode.ProviderResult<vscode.CodeLens[]> {
     if (document.fileName.endsWith("template.json")) {
+      // env info needs to be reloaded
+      this.projectConfigs = undefined;
       return this.computeTemplateCodeLenses(document);
     } else {
       return this.computePreviewCodeLenses(document);
     }
   }
 
+  public async resolveCodeLens(
+    lens: vscode.CodeLens,
+    _token: vscode.CancellationToken
+  ): Promise<vscode.CodeLens> {
+    if (isConfigUnifyEnabled() && lens instanceof ManifestPlacholderCodeLens) {
+      const key = lens.placeholder.replace(/{/g, "").replace(/}/g, "");
+      if (!this.projectConfigs) {
+        const release = await this.mutex.acquire();
+        try {
+          if (!this.projectConfigs) {
+            const inputs = getSystemInputs();
+            const getConfigRes = await core.getProjectConfigV3(inputs);
+            if (getConfigRes.isErr()) throw getConfigRes.error;
+            this.projectConfigs = getConfigRes.value;
+          }
+        } finally {
+          release();
+        }
+      }
+
+      if (this.projectConfigs) {
+        let title = "👉";
+        const localEnvInfo = this.projectConfigs.envInfos[environmentManager.getLocalEnvName()];
+        const defaultEnvInfo = this.projectConfigs.envInfos[environmentManager.getDefaultEnvName()];
+
+        const localValue = getPropertyByPath(localEnvInfo, key);
+        title = `${title} ${environmentManager.getLocalEnvName()}: ${localValue}`;
+
+        const defaultValue = getPropertyByPath(defaultEnvInfo, key);
+        title = `${title}, ${defaultEnvInfo.envName}: ${defaultValue}`;
+
+        lens.command = {
+          title: title,
+          command: "fx-extension.openConfigState",
```

Comment:
Does this feature show real value for config?  If so, should pass 'config' as type here.

Reply:
Updated, thanks!